nf_core/lint/files_unchanged.py

```diff
@@ -0,0 +1,202 @@
+        '.gitattributes',
+        '.github/.dockstore.yml',
+        '.github/CONTRIBUTING.md',
+        '.github/PULL_REQUEST_TEMPLATE.md',
+        '.github/markdownlint.yml',
+        '.github/ISSUE_TEMPLATE/bug_report.md',
+        '.github/ISSUE_TEMPLATE/config.yml',
+        '.github/ISSUE_TEMPLATE/feature_request.md',
+        '.github/workflows/awsfulltest.yml',
+        '.github/workflows/awstest.yml',
+        '.github/workflows/branch.yml',
+        '.github/workflows/ci.yml',
+        '.github/workflows/linting.yml',
+        '.github/workflows/linting_comment.yml',
+        '.github/workflows/push_dockerhub_dev.yml',
+        '.github/workflows/push_dockerhub_release.yml',
+        'assets/email_template.html',
+        'assets/email_template.txt',
+        'assets/nf-core-test_logo.png',
+        'assets/sendmail_template.txt',
+        'bin/markdown_to_html.py',
+        'conf/charliecloud.config',
+        'docs/README.md',
+        'docs/images/nf-core-PIPELINE_logo.png'
+
+    Files that can have additional content but must include the template contents::
+
+        '.gitignore',
+        'assets/multiqc_config.yaml'
+
+    .. tip:: You can configure the ``nf-core lint`` tests to ignore any of these checks by setting
+             the ``files_unchanged`` key as follows in your linting config file. For example:
+
+             .. code-block:: yaml
+
+                files_unchanged:
+                  - .github/workflows/branch.yml
+                  - assets/multiqc_config.yaml
+
+    """
+
+    passed = []
+    failed = []
+    ignored = []
+    fixed = []
+
+    # Check that we have the minimum required config
+    try:
+        self.nf_config["manifest.name"]
+        self.nf_config["manifest.description"]
+        self.nf_config["manifest.author"]
+    except KeyError as e:
+        return {"ignored": [f"Required pipeline config not found - {e}"]}
+    short_name = self.nf_config["manifest.name"].strip("\"'").replace("nf-core/", "")
+
+    # NB: Should all be files, not directories
+    # List of lists. Passes if any of the files in the sublist are found.
+    files_exact = [
+        ["CODE_OF_CONDUCT.md"],
+        ["LICENSE", "LICENSE.md", "LICENCE", "LICENCE.md"],  # NB: British / American spelling
+        [".gitattributes"],
+        [os.path.join(".github", ".dockstore.yml")],
+        [os.path.join(".github", "CONTRIBUTING.md")],
+        [os.path.join(".github", "PULL_REQUEST_TEMPLATE.md")],
+        [os.path.join(".github", "markdownlint.yml")],
+        [os.path.join(".github", "ISSUE_TEMPLATE", "bug_report.md")],
+        [os.path.join(".github", "ISSUE_TEMPLATE", "config.yml")],
+        [os.path.join(".github", "ISSUE_TEMPLATE", "feature_request.md")],
+        [os.path.join(".github", "workflows", "awsfulltest.yml")],
+        [os.path.join(".github", "workflows", "awstest.yml")],
+        [os.path.join(".github", "workflows", "branch.yml")],
+        [os.path.join(".github", "workflows", "ci.yml")],
```

Comment:
wouldn't it be desirable to add further tests in the `ci.yml` so it would be a file that must contain the template content but can contain other stuff

Reply:
File removed from this test 👍🏻